x-pack/plugins/triggers_actions_ui/public/application/hooks/use_bulk_edit_select.tsx

```diff
@@ -42,14 +42,15 @@ const reducer = (state: BulkEditSelectionState, action: Action) => {
       };
     case ActionTypes.TOGGLE_ROW: {
       const id = payload as string;
+      const selectedIds = new Set(Array.from(state.selectedIds));
```

Comment:
I'm just curious to know why we needed to make this change?

Reply:
When I started to fix tests, I've noticed that these `selectedIds` are not cleaned from one test to another. 
it's happening because we use const 'initialState' as an initial state:
`const [state, dispatch] = useReducer(reducer, initialState);`
And after that we modified state here: `state.selectedIds.add(id);` and modified `selectedIds` in 'initialState' as well, because they are the same Set.
But it somehow did not repeat in localhost, only tests. That's a mystery for me.
So here I decided to use copy `selectedIds`, not real `selectedIds`.
Does it make sense?

Reply:
I see, if that's the case I think we should just make this change at the start of the hook?
```
  const [state, dispatch] = useReducer(reducer, {
    ...initialState,
    selectedIds: new Set<string>(),
  });
```
Something like this? This way it will fix other problems that may come from using the same reference as the initial state.

Reply:
@JiaweiWu  Fixed here: [create set copy when create reducer](https://github.com/elastic/kibana/pull/151666/commits/9d6e0eb6ab98cb43c733db24e9af00ed89aa9c60)